drivers/video/gc2145.c

```diff
@@ -0,0 +1,1150 @@
+			  struct video_format *fmt)
+{
+	struct gc2145_data *drv_data = dev->data;
+	uint16_t width, height;
+	int ret;
+	int i = 0;
+
+	/* We only support RGB565 formats */
+	if (fmt->pixelformat != VIDEO_PIX_FMT_RGB565) {
+		LOG_ERR("gc2145 camera supports only RGB565");
+		return -ENOTSUP;
+	}
+
+	width = fmt->width;
+	height = fmt->height;
+
+	if (!memcmp(&drv_data->fmt, fmt, sizeof(drv_data->fmt))) {
+		/* nothing to do */
+		return 0;
+	}
+
+	drv_data->fmt = *fmt;
+
+	/* Set output format */
+	ret = gc2145_set_output_format(dev, fmt->pixelformat);
+	if (ret < 0) {
+		LOG_ERR("Failed to set the output format");
+		return ret;
+	}
+
+	/* Check if camera is capable of handling given format */
+	while (fmts[i].pixelformat) {
+		if (fmts[i].width_min == width && fmts[i].height_min == height &&
+		    fmts[i].pixelformat == fmt->pixelformat) {
+			/* Set window size */
+			ret = gc2145_set_resolution(dev, fmt->width, fmt->height);
+			if (ret < 0) {
+				LOG_ERR("Failed to set the resolution");
+			}
+			return ret;
+		}
+		i++;
+	}
+
+	/* Camera is not capable of handling given format */
+	LOG_ERR("Image format not supported\n");
+	return -ENOTSUP;
+}
+
+static int gc2145_get_fmt(const struct device *dev, enum video_endpoint_id ep,
+			  struct video_format *fmt)
+{
+	struct gc2145_data *drv_data = dev->data;
+
+	*fmt = drv_data->fmt;
+
+	return 0;
+}
+
+static int gc2145_stream_start(const struct device *dev)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	return gc2145_write_reg(&cfg->i2c, 0xf2, 0x0f);
+}
+
+static int gc2145_stream_stop(const struct device *dev)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	return gc2145_write_reg(&cfg->i2c, 0xf2, 0x00);
+}
+
+static int gc2145_get_caps(const struct device *dev, enum video_endpoint_id ep,
+			   struct video_caps *caps)
+{
+	caps->format_caps = fmts;
+	return 0;
+}
+
+static int gc2145_set_ctrl(const struct device *dev, unsigned int cid, void *value)
+{
+	switch (cid) {
+	case VIDEO_CID_HFLIP:
+		return gc2145_set_ctrl_hmirror(dev, (int)value);
+	case VIDEO_CID_VFLIP:
+		return gc2145_set_ctrl_vflip(dev, (int)value);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static const struct video_driver_api gc2145_driver_api = {
+	.set_format = gc2145_set_fmt,
+	.get_format = gc2145_get_fmt,
+	.get_caps = gc2145_get_caps,
+	.stream_start = gc2145_stream_start,
+	.stream_stop = gc2145_stream_stop,
+	.set_ctrl = gc2145_set_ctrl,
+};
+
+static int gc2145_init(const struct device *dev)
+{
+	struct video_format fmt;
+	int ret;
+
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	const struct gc2145_config *cfg = dev->config;
+
+	ret = gpio_pin_configure_dt(&cfg->reset_gpio, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
+	}
+
+	k_sleep(K_MSEC(1));
+	gpio_pin_set_dt(&cfg->reset_gpio, 0);
+	k_sleep(K_MSEC(1));
+#endif
+
+	ret = gc2145_check_connection(dev);
+	if (ret) {
+		return ret;
+	}
+
+	gc2145_soft_reset(dev);
+	k_msleep(300);
```

Comment:
We already have a 30ms sleep in the soft_reset function. You should put the delay either here or in the function.

Reply:
Moved this large delay to the soft_reset routine.